### DISCUSSION TITLE
Webpack +  openpgp and LGPL

### DISCUSSION THREAD

**Discussion by vinodsr** [openpgpjs/openpgpjs#1427] [Q&A]
I am planning to use openpgp with reactJS and nestJS in a commercial project. It is a closed source project . 

Both of the frameworks use webpack for packaging. 

As you know webpack will minify and optimize the code including all libraries and application code. 
I am little concerned about the LGPL3 license of openpgp, where it mentions that only static linking should be done and the source code should not be changed. 

So will using webpack violates the LGPL and i need to publishing the source code of the project.

Thanks in advance. Hope i had made the question clear.

**Comment by twiss** [comment]
Hey :wave: Sorry for the delay. I'm not a lawyer, but to be safe it may be best to keep OpenPGP.js as a separate file, e.g. by doing `const = await import('openpgp');`. That may make it easier to replace OpenPGP.js with a different version, if required.
